packages/babel-plugin/src/utils/ast-builders.tsx

```diff
@@ -526,26 +525,42 @@ export const getPropValue = (
 /**
  * Transforms CSS output into `sheets` and `classNames` ASTs.
  *
- * @param cssOutput {CSSOutput}
+ * @param cssItems {CssItem[]}
  */
-const transformItemCss = (cssOutput: CSSOutput) => {
+const transformItemCss = (cssItems: CssItem[]) => {
   const sheets: string[] = [];
   const classNames: t.Expression[] = [];
 
-  cssOutput.css.forEach((item) => {
-    const css = transformCss(getItemCss(item));
-    const className = css.classNames.join(' ');
+  cssItems.forEach((item) => {
+    switch (item.type) {
+      case 'conditional':
+        const consequent = transformItemCss(item.consequent);
+        const alternate = transformItemCss(item.alternate);
 
-    sheets.push(...css.sheets);
+        sheets.push(...consequent.sheets, ...alternate.sheets);
+        classNames.push(
+          t.conditionalExpression(item.test, consequent.classNames[0], alternate.classNames[0])
```

Comment:
Is it possible for there to be multiple `classNames`? If so, should they be joined together somehow? Or should the types change to not be an array?

Reply:
It shouldn't be possible to have multiple entries in `classNames` here (which is a confusing as the `classNames` array actually represents actual JS Expressions that could contain multiple CSS class names).

I've re-worked the logic here to make that expectation clear up-front when parsing the JS in the first place.